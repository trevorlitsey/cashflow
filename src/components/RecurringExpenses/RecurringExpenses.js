import React from 'react';
import { object, func } from 'prop-types';

import { SubTitle } from '../../styles/SharedComponents';

import RecurringExpenseTable from '../RecurringExpenseTable/RecurringExpenseTable';
import NewRecurringExpenseForm from '../NewRecurringExpenseForm/NewRecurringExpenseForm';

class RecurringExpenses extends React.PureComponent {

	static propTypes = {
		recurringExpenses: object.isRequired,
		addRecurringExpense: func.isRequired,
		deleteRecurringExpense: func.isRequired,
	}

	render() {

		const { recurringExpenses, addRecurringExpense, deleteRecurringExpense, addOneTimeExpense } = this.props;

		return (
			<div>
				<SubTitle>Recurring Income/Expenses:</SubTitle>
				<RecurringExpenseTable recurringExpenses={recurringExpenses} deleteRecurringExpense={deleteRecurringExpense} />
				<NewRecurringExpenseForm addRecurringExpense={addRecurringExpense} />
			</div>
		)
	}
}

export default RecurringExpenses;