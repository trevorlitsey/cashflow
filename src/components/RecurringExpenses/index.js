import React from 'react';
import { object, func } from 'prop-types';
import styled from 'styled-components';

import { SubTitle } from '../../styles/components';

import RecurringExpenseTable from '../RecurringExpenseTable';
import NewRecurringExpenseForm from '../NewRecurringExpenseForm';

class RecurringExpenses extends React.PureComponent {

	static propTypes = {
		recurringExpenses: object.isRequired,
		addRecurringExpense: func.isRequired,
		deleteRecurringExpense: func.isRequired,
	}

	render() {

		const { recurringExpenses, addRecurringExpense, deleteRecurringExpense, addOneTimeExpense } = this.props;

		return (
			<div>
				<SubTitle>Recurring Income/Expenses:</SubTitle>
				<RecurringExpenseTable recurringExpenses={recurringExpenses} deleteRecurringExpense={deleteRecurringExpense} />
				<NewRecurringExpenseForm addRecurringExpense={addRecurringExpense} />
			</div>
		)
	}
}

export default RecurringExpenses;