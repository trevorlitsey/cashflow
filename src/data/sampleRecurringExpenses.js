const recurringExpenses = {
	one: {
		name: 'ya!',
		startDate: 1522698884359,
		name: 'an expenses',
		frequency: 2,
		interval: 'days',
	},
	two: {
		name: 'check this out',
		startDate: 1522698884359,
		name: 'an expenses',
		frequency: 1,
		interval: 'weeks',
	}
}

export default recurringExpenses;