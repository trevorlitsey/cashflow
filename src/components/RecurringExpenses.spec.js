import React from 'react';
import { shallow } from 'enzyme'

import sampleRecurringExpenses from '../data/sampleRecurringExpenses';

import RecurringExpenses from './RecurringExpenses';

describe('RecurringExpenses', () => {

	it('should render without crashing', () => {
		const wrapper = renderRecurringExpenses();
		expect(wrapper).toMatchSnapshot();
	})

})

function renderRecurringExpenses(props = {}) {
	const propsToUser = {
		recurringExpenses: sampleRecurringExpenses,
		addRecurringExpense: () => { },
		deleteRecurringExpense: () => { },
		addOneTimeExpense: () => { },
		...props,
	}
	return shallow(<RecurringExpenses {...propsToUser} />)
}