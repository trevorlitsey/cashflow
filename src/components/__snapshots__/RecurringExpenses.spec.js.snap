// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecurringExpenses should render without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RecurringExpenses
    addOneTimeExpense={[Function]}
    addRecurringExpense={[Function]}
    deleteRecurringExpense={[Function]}
    recurringExpenses={
      Object {
        "one": Object {
          "frequency": 2,
          "interval": "days",
          "name": "an expenses",
          "startDate": 1522698884359,
        },
        "two": Object {
          "frequency": 1,
          "interval": "weeks",
          "name": "an expenses",
          "startDate": 1522698884359,
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <styled.h3>
          Recurring Income/Expenses:
        </styled.h3>,
        <RecurringExpenseTable
          deleteRecurringExpense={[Function]}
          recurringExpenses={
            Object {
              "one": Object {
                "frequency": 2,
                "interval": "days",
                "name": "an expenses",
                "startDate": 1522698884359,
              },
              "two": Object {
                "frequency": 1,
                "interval": "weeks",
                "name": "an expenses",
                "startDate": 1522698884359,
              },
            }
          }
        />,
        <NewRecurringExpenseForm
          addRecurringExpense={[Function]}
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Recurring Income/Expenses:",
        },
        "ref": null,
        "rendered": "Recurring Income/Expenses:",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "deleteRecurringExpense": [Function],
          "recurringExpenses": Object {
            "one": Object {
              "frequency": 2,
              "interval": "days",
              "name": "an expenses",
              "startDate": 1522698884359,
            },
            "two": Object {
              "frequency": 1,
              "interval": "weeks",
              "name": "an expenses",
              "startDate": 1522698884359,
            },
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "addRecurringExpense": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <styled.h3>
            Recurring Income/Expenses:
          </styled.h3>,
          <RecurringExpenseTable
            deleteRecurringExpense={[Function]}
            recurringExpenses={
              Object {
                "one": Object {
                  "frequency": 2,
                  "interval": "days",
                  "name": "an expenses",
                  "startDate": 1522698884359,
                },
                "two": Object {
                  "frequency": 1,
                  "interval": "weeks",
                  "name": "an expenses",
                  "startDate": 1522698884359,
                },
              }
            }
          />,
          <NewRecurringExpenseForm
            addRecurringExpense={[Function]}
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Recurring Income/Expenses:",
          },
          "ref": null,
          "rendered": "Recurring Income/Expenses:",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "deleteRecurringExpense": [Function],
            "recurringExpenses": Object {
              "one": Object {
                "frequency": 2,
                "interval": "days",
                "name": "an expenses",
                "startDate": 1522698884359,
              },
              "two": Object {
                "frequency": 1,
                "interval": "weeks",
                "name": "an expenses",
                "startDate": 1522698884359,
              },
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "addRecurringExpense": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
